src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java

```diff
@@ -55,133 +54,144 @@
-        Subscription subscription = subscriptions.get(0);
+    @CrossOrigin
+    @ApiOperation(value = "Updates the existing subscriptions")
+    public ResponseEntity<List<SubscriptionResponse>> updateSubscriptions(@RequestBody List<Subscription> subscriptions) {
+        errorMap = new HashMap<>();
         String user = (authenticate) ? HttpSessionConfig.getCurrentUser() : "";
-        subscription.setUserName(user);
-        String subscriptionName = subscription.getSubscriptionName();
-        LOG.info("Subscription :" + subscriptionName + " update started");
-        SubscriptionResponse subscriptionResponse = new SubscriptionResponse();
-        try {
-            subscription.setCreated(Instant.now().toEpochMilli());
-            subscriptionValidator.validateSubscription(subscription);
-        } catch (SubscriptionValidationException e) {
-            String msg = "Validation of Subscription parameters on:" + subscription.getSubscriptionName()
-                    + " failed! Error: " + e.getMessage();
-            LOG.error(msg);
-            subscriptionResponse.setMsg(msg);
-            subscriptionResponse.setStatusCode(HttpStatus.PRECONDITION_FAILED.value());
-            return new ResponseEntity<>(subscriptionResponse, HttpStatus.PRECONDITION_FAILED);
-        }
 
-        if (subscriptionService.doSubscriptionExist(subscriptionName)) {
-            subscriptionService.modifySubscription(subscription, subscriptionName);
-            LOG.info("Subscription :" + subscriptionName + " update completed");
-            subscriptionResponse.setMsg("Updated Successfully");
-            subscriptionResponse.setStatusCode(HttpStatus.OK.value());
-            return new ResponseEntity<>(subscriptionResponse, HttpStatus.OK);
-
-        } else {
-            LOG.error("Subscription :" + subscription.getSubscriptionName() + " can't be found.");
-            subscriptionResponse.setMsg("Subscription can't be found");
-            subscriptionResponse.setStatusCode(HttpStatus.BAD_REQUEST.value());
-            return new ResponseEntity<>(subscriptionResponse, HttpStatus.BAD_REQUEST);
+        subscriptions.forEach(subscription -> {
+            String subscriptionName = subscription.getSubscriptionName();
+            LOG.debug("Subscription updating has been started: " + subscriptionName);
+            try {
+                subscriptionValidator.validateSubscription(subscription);
+
+                if (subscriptionService.doSubscriptionExist(subscriptionName)) {
+                    subscription.setUserName(user);
+                    subscription.setCreated(Instant.now().toEpochMilli());
+                    subscriptionService.modifySubscription(subscription, subscriptionName);
+                    LOG.debug("Subscription updating is completed: " + subscriptionName);
+                } else {
+                    LOG.error("Subscription is not found: " + subscriptionName);
```

Comment:
Update message here to "Subscription to update is not found". Use a local variable too since same text is used for both logging and errorMap.